File: js/stage3/S3_OBJ_06.js
// S3_OBJ_06 — Group by
function groupBy(items, key) {
  const result = {};
  for (const item of items) {
    const value = item[key];
    if (!result[value]) result[value] = [];
    result[value].push(item);
  }
  return result;
}
// Test
console.log(
  groupBy(
    [
      { name: "A", type: "x" },
      { name: "B", type: "y" },
      { name: "C", type: "x" },
    ],
    "type"
  )
);

File: Python/stage3/S3_OBJ_06.py
# S3_OBJ_06 — Group by key
def group_by(items, key):
   result = {}
   for item in items:
       k = item.get(key)
       result.setdefault(k, []).append(item)
   return result
items = [
   {"type": "a", "value": 1},
   {"type": "b", "value": 2},
   {"type": "a", "value": 3},
]
print(group_by(items, "type"))